src/test/java/uk/nhs/hee/tis/trainee/forms/api/validation/FormRPartBValidatorTest.java

```diff
@@ -0,0 +1,187 @@
+    FormRPartB formRPartB1 = new FormRPartB();
+    formRPartB1.setTraineeTisId(DEFAULT_TRAINEE_TIS_ID);
+    formRPartB1.setId("ANOTHER_ID_1");
+    formRPartB1.setLifecycleState(LifecycleState.DRAFT);
+
+    FormRPartB formRPartB2 = new FormRPartB();
+    formRPartB2.setTraineeTisId(DEFAULT_TRAINEE_TIS_ID);
+    formRPartB2.setId("ANOTHER_ID_2");
+    formRPartB2.setLifecycleState(LifecycleState.DRAFT);
+
+    when(formRPartBRepositoryMock
+        .findByTraineeTisIdAndLifecycleState(formRPartBDto.getTraineeTisId(),
+            formRPartBDto.getLifecycleState()))
+        .thenReturn(Lists.list(formRPartB1, formRPartB2));
+
+    List<FieldError> fieldErrors = validator.checkIfDraftUnique(formRPartBDto);
+    assertThat("Should return 1 error", fieldErrors.size(), is(1));
+    assertThat("Error not valid", fieldErrors.get(0).getDefaultMessage(),
+        is("More than one draft form R Part B already exist"));
+  }
+
+  @Test
+  void validateUpdateDraftIfOneDraftWithSameIdFound() {
+    FormRPartB formRPartB = new FormRPartB();
+    formRPartB.setTraineeTisId(DEFAULT_TRAINEE_TIS_ID);
+    formRPartB.setId(DEFAULT_ID);
+    formRPartB.setLifecycleState(LifecycleState.DRAFT);
+
+    when(formRPartBRepositoryMock
+        .findByTraineeTisIdAndLifecycleState(formRPartBDto.getTraineeTisId(),
+            formRPartBDto.getLifecycleState()))
+        .thenReturn(Lists.list(formRPartB));
+
+    List<FieldError> fieldErrors = validator.checkIfDraftUnique(formRPartBDto);
+    assertThat("Should not return any errors", fieldErrors.size(), is(0));
+  }
+
+  @Test
+  void validateUpdateDraftIfOneDraftWithDifferentIdFound() {
+    FormRPartB formRPartB = new FormRPartB();
+    formRPartB.setTraineeTisId(DEFAULT_TRAINEE_TIS_ID);
+    formRPartB.setId("ANOTHER_ID");
+    formRPartB.setLifecycleState(LifecycleState.DRAFT);
+
+    when(formRPartBRepositoryMock
+        .findByTraineeTisIdAndLifecycleState(formRPartBDto.getTraineeTisId(),
+            formRPartBDto.getLifecycleState()))
+        .thenReturn(Lists.list(formRPartB));
+
+    List<FieldError> fieldErrors = validator.checkIfDraftUnique(formRPartBDto);
+    assertThat("Should return 1 error", fieldErrors.size(), is(1));
+    assertThat("Error not valid", fieldErrors.get(0).getDefaultMessage(),
+        is("Draft form R Part B already exists"));
+  }
+
+  @Test
+  void validateCreateDraftIfOneDraftFound() {
+    formRPartBDto.setId(null);
+
+    FormRPartB formRPartB = new FormRPartB();
+    formRPartB.setTraineeTisId(DEFAULT_TRAINEE_TIS_ID);
+    formRPartB.setId("ANOTHER_ID");
+    formRPartB.setLifecycleState(LifecycleState.DRAFT);
+
+    when(formRPartBRepositoryMock
+        .findByTraineeTisIdAndLifecycleState(formRPartBDto.getTraineeTisId(),
+            formRPartBDto.getLifecycleState()))
+        .thenReturn(Lists.list(formRPartB));
+
+    List<FieldError> fieldErrors = validator.checkIfDraftUnique(formRPartBDto);
+    assertThat("Should not return any errors", fieldErrors.size(), is(1));
```

Comment:
I think this message is the only thing to change

Reply:
Sorry, my bad, again. Thought I missed one last time. Fixed them both.

Reply:
Ooops.  I missed the one in PartA too!